rest-api/dao/physical_measurements_dao.py

```diff
@@ -291,32 +291,49 @@ def _update_participant_summary(self, session, obj):
       raise BadRequest("Can't submit physical measurements for participant %s without consent" %
                        participant_id)
     raise_if_withdrawn(participant_summary)
-    participant_summary.physicalMeasurementsTime = obj.created
-    participant_summary.physicalMeasurementsFinalizedTime = obj.finalized
-    participant_summary.physicalMeasurementsCreatedSiteId = obj.createdSiteId
-    participant_summary.physicalMeasurementsFinalizedSiteId = obj.finalizedSiteId
     participant_summary.lastModified = clock.CLOCK.now()
 
-    if obj.status and obj.status == PhysicalMeasurementsStatus.CANCELLED:
-      if not self.has_uncancelled_pm(session, participant):
-        # update corresponding participant summary physicalMeasurementsStatus
-        participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.CANCELLED
-        participant_summary.physicalMeasurementsTime = None
+    # These fields set on measurement that is cancelled and doesn't have a previous good measurement
+    if obj.status and obj.status == PhysicalMeasurementsStatus.CANCELLED and not \
+       self.has_uncancelled_pm(session, participant):
 
-    if participant_summary.physicalMeasurementsStatus != PhysicalMeasurementsStatus.CANCELLED or \
-       obj.status != PhysicalMeasurementsStatus.CANCELLED:
-      # if a PM was restored, it is complete again.
+      participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.CANCELLED
+      participant_summary.physicalMeasurementsTime = None
+
+    # These fields set on any measurement not cancelled
+    elif obj.status != PhysicalMeasurementsStatus.CANCELLED:
+      # new PM or if a PM was restored, it is complete again.
       participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.COMPLETED
+      participant_summary.physicalMeasurementsTime = obj.created
+      participant_summary.physicalMeasurementsFinalizedTime = obj.finalized
+      participant_summary.physicalMeasurementsCreatedSiteId = obj.createdSiteId
+      participant_summary.physicalMeasurementsFinalizedSiteId = obj.finalizedSiteId
+
+    if obj.status and obj.status == PhysicalMeasurementsStatus.CANCELLED and \
```

Comment:
made elif, we do need it because we changed these fields to None when cancelled, but if they have a previous good PM we don't want participant summary to reflect these changes.

Reply:
OK, got it.